File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        
        int l = 0;
        int r = numbers.size()-1;
        
        vector<int> result;
        
        while(l<r){
            
            int curSum = numbers[l] + numbers[r];
            
            if(curSum > target){
                r--;
            }
            else if(curSum < target){
                l++;
            }
            else{
                result.push_back(l+1);
                result.push_back(r+1);
                return result;
            }
            
        }
        return result;
        
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
## time:o(n)
## space:o(n)
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        
        ##create two pointers
        
        l , r = 0, len(numbers)-1
        
        while(l<r):
            
            # calculate the current sum at pointers
            curSum = numbers[l] + numbers[r]
            
            ## if sum is greater than target decreament right pointer because the array is in sorted order
            if(curSum > target):
                r-=1
                
            ## if sum is less than target decreament left pointer because the array is in sorted order    
            elif(curSum < target):
                l+=1
                
            else:
                return [l+1 , r+1]
            
        return []   
        